packages/radix-vue/src/shared/useBodyScrollLock.ts

```diff
@@ -7,93 +7,98 @@ import { nextTick, onBeforeUnmount, ref, watch } from 'vue'
 import { defu } from 'defu'
 import { injectConfigProviderContext } from '@/ConfigProvider/ConfigProvider.vue'
 
-const useInitialOverflowStyle = createGlobalState(() => ref<string | undefined>())
-const useBodyLockStackCount = createGlobalState(() => ref(0))
+const useBodyLockStackCount = createGlobalState(() => {
+  const count = ref(0)
```

Comment:
also, I would like to keep `createGlobalState` as a state only reactive value, and not incurring side-effect (watcher) inside the function. 

Let move the function/watcher out into the `useBodyScrollLock` composable

Reply:
Okay I got it working with `createSharedComposable`, I think it's working well now. However while writing the tests and having to mock createSharedComposable made me realize that the implementation could lead to bugs if multiple ConfigProviders are used through the same app. Because the `injectConfigProviderContext` call is inside the `createSharedComposable`, it will only be called once and therefore, if multiple Config Providers exists only the first value will be used.

This is why in the tests I mocked `createSharedComposable` to disable caching and just return the function. So here is the question, is using multiple Config Providers an expected use case or should it be considered an anti-pattern?

Switching to a shared composable style was something neat I wanted to add but isn't strictly needed for this PR which its goal was to open useBodyScrollLock to the outside, which could be done by keeping the code closer to how it used to be.

Reply:
`createSharedComposable` make sense!